### PR TITLE
Add DiracSplitter for using DiracFiles locally

### LINKED ISSUE #945: Local() backend doesn't cope with inaccessible LFNs
If my job has splitter `SplitByFiles(ignoremissing=False)` and uses the `Dirac()` backend, then if some of the input `DiracFile`s don't have any accessible replicas (because a site is down) I get an error.

If I use the `Local()` backend for the same job then there is no error, the subjobs corresponding to LFNs without accessible replicas are just run with no inputdata, i.e. `__GangaInputData.txt__` is empty.

It would be better if the `Local()` backend threw an error as the `Dirac()` one does.

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #945 
This adds a check for DiracFiles in the dataset when splitting and not using the Dirac backend. If they are all DiracFiles then it uses the `OfflineGangaDiracSplitter' in order to check for the availability of all the files.

I also found a bug in LHCbDataset  - I wasn't sure of the logic that the code was after but I think this fixes it.

### CODE REVIEW

File: python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py [hunk 1]
```diff
@@ -83,7 +83,8 @@ def isFileTest(_file):
                 for this_file in files:
                     self.files.append(deepcopy(this_file))
             elif isType(files, IGangaFile):
-                self.files.append(deepcopy(this_file))
+                for this_file in files:
```
Comment (rob-c) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: Are we sure this does what we want for all IGangaFiles without subfiles?

Comment (mesmith75) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: ah, I think I may have misinterpreted what it was trying to do here.

Comment (mesmith75) [inline python/GangaLHCb/Lib/LHCbDataset/LHCbDataset.py]: I guess IGangaFiles will be in the GangaList below - so this is just for a single file?

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
retest this please

**Comment by mesmith75** [comment]
retest this please

**Pull request merged**